src/state/psarstake/hederaHooks.ts

```diff
@@ -0,0 +1,337 @@
+import { hederaFn } from 'src/utils/hedera';
+import { maxAmountSpend } from 'src/utils/maxAmountSpend';
+import { useSingleContractMultipleData } from '../pmulticall/hooks';
+import { useDerivedStakeInfo } from '../pstake/hooks';
+import { useTransactionAdder } from '../ptransactions/hooks';
+import { useTokenBalance } from '../pwallet/hooks';
+import { Position, URI } from './types';
+
+/**
+ *
+ * @param positionId Id of a Position
+ * @returns Return some utils functions for stake more or create a new Position
+ */
+export function useDerivativeHederaSarStake(positionId?: BigNumber) {
+  const [attempting, setAttempting] = useState(false);
+  const [hash, setHash] = useState<string | null>(null);
+  const [typedValue, setTypedValue] = useState('');
+  const [stepIndex, setStepIndex] = useState(4);
+  const [stakeError, setStakeError] = useState<string | null>(null);
+
+  const { account } = usePangolinWeb3();
+  const chainId = useChainId();
+
+  const sarStakingContract = useSarStakingContract();
+  const sarNftContract = useSarNFTStakingContract();
+
+  const addTransaction = useTransactionAdder();
+  const { t } = useTranslation();
+
+  const png = PNG[chainId];
+
+  const userPngBalance = useTokenBalance(account ?? ZERO_ADDRESS, png);
+
+  // used for max input button
+  const maxAmountInput = maxAmountSpend(chainId, userPngBalance);
+
+  const useUSDCPrice = useUSDCPriceHook[chainId];
+  const usdcPrice = useUSDCPrice(png);
+  const dollerWorth =
+    userPngBalance?.greaterThan('0') && usdcPrice ? Number(typedValue) * Number(usdcPrice.toFixed()) : undefined;
+
+  const wrappedOnDismiss = useCallback(() => {
+    setStakeError(null);
+    setTypedValue('');
+    setStepIndex(0);
+    setHash(null);
+    setAttempting(false);
+  }, []);
+
+  const { parsedAmount, error } = useDerivedStakeInfo(typedValue, png, userPngBalance);
+  const [approval, approveCallback] = useHederaApproveCallback(chainId, parsedAmount, sarStakingContract?.address);
+
+  const mixpanel = useMixpanel();
+
+  const onUserInput = useCallback((_typedValue: string) => {
+    setTypedValue(_typedValue);
+  }, []);
+
+  const handleMax = useCallback(() => {
+    maxAmountInput && onUserInput(maxAmountInput.toExact());
+    setStepIndex(4);
+  }, [maxAmountInput, onUserInput]);
+
+  const onChangePercentage = (value: number) => {
+    if (!userPngBalance) {
+      setTypedValue('0');
+      return;
+    }
+    if (value === 100) {
+      setTypedValue((userPngBalance as TokenAmount).toExact());
+    } else if (value === 0) {
+      setTypedValue('0');
+    } else {
+      const newAmount = (userPngBalance as TokenAmount)
+        .multiply(JSBI.BigInt(value))
+        .divide(JSBI.BigInt(100)) as TokenAmount;
+
+      setTypedValue(newAmount.toSignificant(6));
+    }
+  };
+
+  const {
+    associate: associatePNG,
+    hederaAssociated: isAssociatedPNG,
+    isLoading: isLoadingPNG,
+  } = useHederaTokenAssociated(PNG[chainId]);
```

Comment:
i think you don't need to check whether PBAR is associated or not because user can only stake if user has PBAR and user can have PBAR only if its associated.